src/admin/reducers/UserReducer.ts

```diff
@@ -23,12 +23,34 @@ import * as  UserActionsCreator  from 'admin/actions/UserActionsCreator';
 import AdminConstants from 'admin/constants/AdminConstants';
 
 const initialState = Immutable.from<User>({
-	name: '',
-	email: '',
-	role: UserRole.Journalist,
+	name: localStorage.getItem('rcUsername') || '',
+	email: localStorage.getItem('rcUserEmail') || '',
+	role: getRoleFromEnum(),
 });
 
+function getRoleFromEnum() {
+	let role: UserRole;
+	switch (localStorage.getItem('rcUserRole')) {
+		case UserRole.Editor:
+			role = UserRole.Editor;
+			break;
+		case UserRole.SiteAdmin:
+			role = UserRole.SiteAdmin;
+			break;
+		case UserRole.SystemAdmin:
+			role = UserRole.SystemAdmin;
+			break;
+		default:
+			role = UserRole.Journalist;
+			break;
+	}
+	return role;
+}
+
 function updateUser(action, state) {
+	localStorage.setItem('rcUsername', action.payload.name);
```

Comment:
move it to Action, and all localStorage operations in deauthenticateUser func as well.
Reducer is only for operation with state

Reply:
Done